Move derive macros docs to scylla crate

Lorak-mmk opened the pull request:
This PR moves docs for serialization / deserialization related derive macros from scylla-cql crate to scylla crate.
This is done to work around the issue that this documentation was not rendered correctly on docs.rs for scylla crate.
It's hard to tell why that is exactly - it's probably a bug / limitation in rustdoc.
It looks like when we have item X in crate A, re-export X as Y in crate B and document it, and re-export Y az Z in crate C,
Y's docs won't be rendered in crate C, only X's and Z's will.

Fixes: #904 

## Pre-review checklist

<!--
    Make sure you took care of the issues on the list.
    Put 'x' into those boxes which apply.
    You can also create the PR now and click on all relevant checkboxes.
    See CONTRIBUTING.md for more details.
-->

- [ ] I have split my patch into logically separate commits.
- [ ] All commit messages clearly explain what they change and why.
- [ ] I added relevant tests for new features and bug fixes.
- [ ] All commits compile, pass static checks and pass test.
- [ ] PR description sums up the changes and reasons why they should be introduced.
- [ ] I have provided docstrings for the public items that I want to introduce.
- [ ] I have adjusted the documentation in `./docs/source/`.
- [ ] I added appropriate `Fixes:` annotations to PR description.

Lorak-mmk commented:
After this is merged, we should release a new version, so that users are able to find documentation.

piodul commented on scylla/src/macros.rs:
```diff
@@ -10,7 +10,7 @@ pub use scylla_cql::macros::FromUserType;
 /// Works only on simple structs without generics etc
 pub use scylla_cql::macros::IntoUserType;
 
-/// Derive macro for the [`SerializeCql`](crate::types::serialize::value::SerializeCql) trait
```
Nit (last commit): typo in the commit message: `scyll` -> `scylla`

piodul commented on scylla-macros/src/lib.rs:
```diff
@@ -9,7 +9,10 @@ mod value_list;
+///
+/// Documentation for this macro can only be found
+/// in `scylla` crate - not in scylla-macros nor in scylla-cql.
+/// This is because if rustdocs limitations that are hard to explain here.
```
Typo: `if` -> `of`, here and in other docstrings in the same commit

piodul reviewed (changes requested):
While the documentation in the `scylla` crate looks much better, it looks like the docstrings from multiple crates are still being squashed. For example, the message about the macro being documented in the `scylla` crate still appears in `scylla::macros::ScyllaCql`:

![image](https://github.com/scylladb/scylla-rust-driver/assets/4950936/9f8f43a6-ab55-40ab-a615-49c63695802b)

However, I can't find anything on the net about overriding docstrings on reexport, so I can't suggest any way to get rid of them (if it is possible at all). As we discussed on slack, let's at least make the message from `scylla-macros` more separated from the rest by putting a separator at the end of the docstrings in `scylla`.

Lorak-mmk commented:
v2
- Addresed review comments (fixed typos, added separators in scylla crate instead of empty lines in scylla-macros)
- Added empty lines in documentation of old macros, so that only the first line is shown in the summary in the main page of documentation.
- Fixed formatting of commit messages
